### PR TITLE
adding traceroute info 

### PR DESCRIPTION
**Pull request by Chr1st0ph3rTurn3r**

### CODE REVIEW

File: docs/ts_traceroute.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+title: Traceroute
+sidebar_label: Traceroute
+---
+
```
Comment (MichaelBaj) [inline docs/ts_traceroute.md:5]: I thought we had a precedent for indicating the version of software a feature was introduced, but I couldn't find any consistency in our docs.
Please add a section at the beginning of this doc indicating the version of software where this was first introduced.

**Review by MichaelBaj — CHANGES_REQUESTED** [review]

File: docs/config_reference_guide.md [hunk 2]
```diff
@@ -2925,6 +2925,37 @@ This sub-element allows administrators to control the default BGP timers the SSR
...
+
+#### Description
+
+The element allows operators to debug their connectivity from point to point.
+
+The `traceroute` command creates a traceroute context for probing the path to a specified endpoint. Traceroute targets can be specified as SVR peers, service/tenant defined endpoints, or targets reachable by routing.
+
+| Element | Type | Description |
+| --- | --- | --- |
+| destination-ip | IP address | The destination address for the traceroute command |
```
Comment (dmontgomeryJuniper) [inline docs/config_reference_guide.md:2942]: destination-ip is now a positional rather than keyword argument

File: docs/ts_traceroute.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+Traceroute is implemented as a troubleshooting tool, allowing you to debug  connectivity from point to point.
+
+Network administrators frequently make use of Linux traceroute as a compliment to ping when isolating intermediate connectivity issues and troubleshooting MTU, and TTL issues. Traceroute provides information on the intermediate connectivity on a hop-by-hop basis between a source and destination. The SSR Traceroute commands provide the ability to detect failure points in connectivity along the network traffic flow. With the routing model and distributed nature of the SSR, traceroute behaves differently depending on the use case.
```
Comment (dmontgomeryJuniper) [inline docs/ts_traceroute.md:8]: *complement

File: docs/ts_traceroute.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+- `routed traceroute`: Mirrors the traceroute utility in Unix. Uses the service, tenant, and interface according to the SSR’s route table.
+- `service traceroute`: Analogous to the service-ping functionality. Useful to debug test traffic as defined in the deployment.
+- `peer traceroute`: Performs traceroute between two adjacent SSRs. Useful for diagnosing peering issues.
+
+## How it Works
+
+Discovery packets are sent from the originating SSR to a single the destination port with varying time-to-live (TTL) values. For each iteration, the TTL is increased by 1, and a 3 packet burst is transmitted. This process is repeated until the originating SSR receives a suitable ‘terminating’ packet from the final destination, or until an upper-limit is reached for hop counts. 
+
+For service-traceroute and peer-traceroute, ICMP echo requests are the default discovery packets. To support instances where ICMP replies are disabled at the destination or disallowed by firewall rules, UDP and TCP are also supported.
```
Comment (dmontgomeryJuniper) [inline docs/ts_traceroute.md:20]: we're only doing UDP

**Review by MichaelBaj — APPROVED** [review]

File: docs/config_reference_guide.md [hunk 5]
```diff
@@ -2925,6 +2925,37 @@ This sub-element allows administrators to control the default BGP timers the SSR
 | --- | --- |
 | 1.0.0 | Introduced |
 
+## traceroute
+
+#### Path 
+
+authority > router > traceroute
```
Comment (dmontgomeryJuniper) [inline docs/config_reference_guide.md:2932]: just hit me: this... isn't a config element.  is this the right guide?

**Review by MichaelBaj — APPROVED** [review]

**Review by smcculley-juniper — APPROVED** [review]

File: docs/cli_reference.md [hunk 6]
```diff
@@ -10368,9 +10368,52 @@ trace [verbose] [verbose] [verbose] [no-logs] <command> [<command> ...]
...
+#### Usage 
+```
+traceroute [peer-name <peer-name>][source-ip <source-ip>] [service <service>] [tenant <tenant>] destination-ip <destination-ip>
+```
+
+##### Keyword Arguments
+
+| name | description |
+| ---- | ----------- |
+| peer-name | The name of the SVR peer to probe |
```
Comment (dmontgomeryJuniper) [inline docs/cli_reference.md:10382]: (default is empty)

File: docs/ts_traceroute.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+admin@combo-west-a.combo-west# traceroute
+usage: traceroute [peer-name <peer-name>][source-ip <source-ip>] [service <service>] [tenant <tenant>] destination-ip <destination-ip>
+```
+
+The `traceroute` command creates a traceroute context for probing the path to a specified endpoint. Traceroute targets can be specified as SVR peers, service/tenant defined endpoints, or targets reachable by routing.
+
+| Element | Type | Description |
+| --- | --- | --- |
+| destination-ip | positional | The destination IP address for the traceroute command |
+| peer-name | string | The name of the SVR peer to probe |
```
Comment (dmontgomeryJuniper) [inline docs/ts_traceroute.md:39]: add (default is empty)

**Review by dmontgomeryJuniper — APPROVED** [review]

**Comment by Chr1st0ph3rTurn3r** [comment]
Closing as it has been merged into the 6.1.x branch.

**Pull request closed without merging**